test/test_db.py

```diff
@@ -39,40 +46,66 @@ def teardown_function(function):
-        # Unicode, just in case
-        [None, 'embölaliå', Identifier('EmbölaliÅ')],
-    ]
-
-    for test in tests:
-        test[0] = db.get_nick_id(test[2], create=True)
-        nick_id, slug, nick = test
-        registered = session.query(Nicknames) \
-                            .filter(Nicknames.canonical == nick) \
-                            .all()
-        assert len(registered) == 1
-        assert registered[0].slug == slug and registered[0].canonical == nick
-
-    # Check that each nick ended up with a different id
-    assert len(set(test[0] for test in tests)) == len(tests)
+    """Test get_nick_id does not create NickID by default."""
+    nick = Identifier('Exirel')
+    session = db.session()
+
+    # Attempt to get NickIDs: it is not created by default
+    with pytest.raises(ValueError):
+        db.get_nick_id(nick)
+
+    # Create the NickIDs
+    nick_id = db.get_nick_id(nick, create=True)
+
+    # Check that one and only one Nicknames exist with that ID
+    nickname = session.query(Nicknames).filter(
+        Nicknames.nick_id == nick_id,
+    ).one()  # will raise if not one and exactly one
+    assert nickname.canonical == 'Exirel'
+    assert nickname.slug == nick.lower()
+
+    session.close()
+
+
+@pytest.mark.parametrize('name, slug, variant', (
+    # Check case insensitive with ASCII only
+    ('Embolalia', 'embolalia', 'eMBOLALIA'),
+    # Ensures case conversion is handled properly
+    ('[][]', '{}{}', '[}{]'),
+    # Unicode, just in case
+    ('EmbölaliÅ', 'embölaliÅ', 'EMBöLALIÅ'),
+))
+def test_get_nick_id_casemapping(db, name, slug, variant):
+    """Test get_nick_id is case-insensitive through an Identifier."""
+    session = db.session()
+    nick = Identifier(name)
+
+    # Create the NickIDs
+    nick_id = db.get_nick_id(nick, create=True)
+
+    registered = session.query(Nicknames) \
+                        .filter(Nicknames.canonical == name) \
+                        .all()
+    assert len(registered) == 1
+    assert registered[0].slug == slug
+    assert registered[0].canonical == name
 
     # Check that the retrieval actually is idempotent
-    for test in tests:
-        nick_id = test[0]
-        # no `create` this time, since the ID should already exist
-        new_id = db.get_nick_id(test[2])
-        assert nick_id == new_id
+    assert nick_id == db.get_nick_id(name)
 
     # Even if the case is different
-    for test in tests:
-        nick_id = test[0]
-        # still no `create`, because the nick ID should already exist now
-        new_id = db.get_nick_id(Identifier(test[2].upper()))
-        assert nick_id == new_id
+    assert nick_id == db.get_nick_id(variant)
+
+    # And no other NickIDs was created (even with create=True)
+    assert nick_id == db.get_nick_id(name, create=True)
+    assert nick_id == db.get_nick_id(variant, create=True)
+    assert 1 == session.query(NickIDs).count()
+
+    # But a very different name means a new NickIDs
```

Comment:
```suggestion
    # But a truly different name means a new NickID
```

Now I've added "splitting semantic hairs" to the ongoing battle against the nemesi**S** 😀